Wangscape/utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <type_traits>
+#include <stdexcept>
+
+namespace
+{
+
+template<typename I>
+I _pow(I base, I exp)
+{
+    I result = 1;
+    for (int i = 0; i < std::numeric_limits<I>::digits; i++)
+    {
+        if (exp & 1)
+            result *= base;
+        else if (exp == 0)
+            break;
+        exp >>= 1;
+        base *= base;
+    }
+    return result;
+}
+
+}
+
+template <typename I>
+typename std::enable_if<std::is_unsigned<I>::value, I>::type pow(I base, I exp)
```

Comment:
`pow(float, float)` and `pow(double, double)` seem to be in the global namespace, not just `std`, and ambiguous calls like `int i; size_t s; pow(i, s);` cause the compiler to try to match the `std` version instead of this. This should be renamed to something like `ipow`.